### PR TITLE
Upload: Add more chunk resuming unit tests

### PR DESCRIPTION
**Pull request by ckamm**
For testing #6405

### CODE REVIEW

File: src/libsync/propagateuploadng.cpp [hunk 1]
```diff
@@ -219,7 +224,9 @@ void PropagateUploadFileNG::slotDeleteJobFinished()
 
 void PropagateUploadFileNG::startNewUpload()
 {
-    ASSERT(propagator()->_activeJobList.count(this) == 1);
+    if (propagator()->_activeJobList.count(this) == 0)
```
Comment (ogoffart) [inline src/libsync/propagateuploadng.cpp]: So you are changing the invariant?  I guess this is only hapenning when there were more data on the server.  But i'd argue that we should be consistant, and always add the activeJobList in the same place.

For example, if we start uploading chunks in parallel again, code like that can lead to inconsistency.

File: test/testchunkingng.cpp [hunk 2]
```diff
@@ -96,6 +98,130 @@ private slots:
...
+                Q_ASSERT(request.rawHeader("OC-Chunk-Offset").toLongLong() >= firstChunk.size);
+            } else if (op == QNetworkAccessManager::DeleteOperation) {
+                deletedPaths.append(request.url().path());
+            }
+            return nullptr;
+        });
+
+        QVERIFY(fakeFolder.syncOnce());
+
+        for (const auto& toDelete : chunksToDelete) {
```
Comment (ogoffart) [inline test/testchunkingng.cpp]: This block of code is not particularly efficient...  But that does not matter :-)

File: test/testchunkingng.cpp [hunk 3]
```diff
@@ -96,6 +98,130 @@ private slots:
...
+        fakeFolder.syncEngine().account()->setCapabilities({ { "dav", QVariantMap{ {"chunking", "1.0"} } } });
+        const int size = 300 * 1000 * 1000; // 300 MB
+        partialUpload(fakeFolder, "A/a0", size);
+        QCOMPARE(fakeFolder.uploadState().children.count(), 1);
+        auto chunkingId = fakeFolder.uploadState().children.first().name;
+        const auto &chunkMap = fakeFolder.uploadState().children.first().children;
+        quint64 uploadedSize = std::accumulate(chunkMap.begin(), chunkMap.end(), 0LL, [](quint64 s, const FileInfo &f) { return s + f.size; });
+        QVERIFY(uploadedSize > 50 * 1000 * 1000); // at least 50 MB
+
+        // Add a chunk that makes the file completely uploaded
```
Comment (ogoffart) [inline test/testchunkingng.cpp]: i'd would have made an error on the move...  but this is fine too.

File: test/testchunkingng.cpp [hunk 4]
```diff
@@ -96,6 +98,130 @@ private slots:
...
+        fakeFolder.setServerOverride([&](QNetworkAccessManager::Operation op, const QNetworkRequest &, QIODevice *) -> QNetworkReply * {
+            if (op == QNetworkAccessManager::PutOperation) {
+                sawPut = true;
+            } else if (op == QNetworkAccessManager::DeleteOperation) {
+                sawDelete = true;
+            }
+            return nullptr;
+        });
+        QVERIFY(!sawPut);
+        QVERIFY(!sawDelete);
```
Comment (ogoffart) [inline test/testchunkingng.cpp]: I guess you meant to put these two QVERIFY after the sync!

(Also maybe add a sawMove... why not)

**Comment by ogoffart** [comment]
Jenkins is failing with:

```
FAIL!  : TestChunkingNG::connectionDroppedBeforeEtagRecieved(small file) '!fakeFolder.syncOnce()' returned FALSE. ()
   Loc: [/var/lib/jenkins/workspace/cloud-client_client_PR-6460-MGEVCM5XTFJQO7UC3CJR3U3LRVCP3QWVQGE5SP3OYDUCM2HY26JA/test/testchunkingng.cpp(553)]
```

I don't know what causes this error though

**Comment by ckamm** [comment]
Updated. The ChunkingNg tests all pass for me locally.

**Pull request merged**